config/m4/fuse3.m4

```diff
@@ -10,21 +10,41 @@ AC_ARG_WITH([fuse3],
             [Enable the use of FUSEv3 (default is guess).])],
             [], [with_fuse3=guess])
 
-AS_IF([test "x$with_fuse3" != xno],
+AC_ARG_WITH([fuse3-static],
+            [AS_HELP_STRING([--with-fuse3-static=(DIR)],
+            [Static link FUSEv3 library (default is no).])],
+            [], [with_fuse3_static=no])
+
+AS_IF([test "x$with_fuse3" != "xno" -o "x$with_fuse3_static" != "xno"],
       [
-       AS_IF([test "x$with_fuse3" = "xguess" \
-                -o "x$with_fuse3" = "xyes" \
-                -o "x$with_fuse3" = "x"],
-             [FUSE3_CPPFLAGS=$(pkg-config --cflags fuse3)
-              FUSE3_LIBS=$(pkg-config --libs fuse3)],
-             [FUSE3_CPPFLAGS="-I${with_fuse3}/include/fuse3"
-              FUSE3_LIBS="-L${with_fuse3}/lib -L${with_fuse3}/lib64"])
+       AS_IF([test "x$with_fuse3_static" != "xno"],
+             [with_fuse3="$with_fuse3_static"])
+
+       save_PKG_CONFIG_PATH="$PKG_CONFIG_PATH"
+
+       AS_IF([test "x$with_fuse3" != "xguess" \
+                -a "x$with_fuse3" != "xyes" \
+                -a "x$with_fuse3" != "x"],
+             [AS_IF([test -d "$with_fuse3/lib64"],
+                    [libsuff="64"], [libsuff=""])
+              ucx_fuse3_pkgconfig="$with_fuse3/lib$libsuff/pkgconfig"
+              export PKG_CONFIG_PATH="$ucx_fuse3_pkgconfig:$PKG_CONFIG_PATH"])
+
+       FUSE3_CPPFLAGS=$(pkg-config --cflags-only-I fuse3)
+       FUSE3_LDFLAGS=$(pkg-config --libs-only-L fuse3)
+       AS_IF([test "x$with_fuse3_static" != "xno"],
+             [FUSE3_LIBS=$(pkg-config --libs-only-l --static fuse3)
+              FUSE3_LIBS=${FUSE3_LIBS#"-lfuse3"}
```

Comment:
seems we rely on pkg-config for CPPFLAGS and LDFLAGS, but have a fallback to "-lfuse" in case of FUSE3_LIBS
need to decide if we require pkg-config (i guess easier is Yes) and if so can remove line 37

Reply:
Simplified lines 37-38